plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/snowflake/TestIcebergSnowflakeCatalogConnectorSmokeTest.java

```diff
@@ -0,0 +1,732 @@
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorBehavior;
+import io.trino.tpch.TpchTable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.sql.SQLException;
+
+import static io.trino.plugin.iceberg.IcebergTestUtils.checkParquetFileSorting;
+import static io.trino.plugin.iceberg.catalog.snowflake.TestingSnowflakeServer.SNOWFLAKE_JDBC_URI;
+import static io.trino.plugin.iceberg.catalog.snowflake.TestingSnowflakeServer.SNOWFLAKE_PASSWORD;
+import static io.trino.plugin.iceberg.catalog.snowflake.TestingSnowflakeServer.SNOWFLAKE_ROLE;
+import static io.trino.plugin.iceberg.catalog.snowflake.TestingSnowflakeServer.SNOWFLAKE_TEST_DATABASE;
+import static io.trino.plugin.iceberg.catalog.snowflake.TestingSnowflakeServer.SNOWFLAKE_TEST_SCHEMA;
+import static io.trino.plugin.iceberg.catalog.snowflake.TestingSnowflakeServer.SNOWFLAKE_USER;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static org.apache.iceberg.FileFormat.PARQUET;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+public class TestIcebergSnowflakeCatalogConnectorSmokeTest
+        extends BaseIcebergConnectorSmokeTest
+{
+    private static final String NATION = TpchTable.NATION.getTableName();
+    private static final String REGION = TpchTable.REGION.getTableName();
+
+    public static final String S3_ACCESS_KEY = requireSystemProperty("testing.snowflake.catalog.s3.access-key");
+    public static final String S3_SECRET_KEY = requireSystemProperty("testing.snowflake.catalog.s3.secret-key");
+    public static final String S3_REGION = requireSystemProperty("testing.snowflake.catalog.s3.region");
+    public static final String SNOWFLAKE_S3_EXTERNAL_VOLUME = requireSystemProperty("testing.snowflake.catalog.s3.external.volume");
+
+    private TestingSnowflakeServer server;
+
+    public TestIcebergSnowflakeCatalogConnectorSmokeTest()
+    {
+        super(PARQUET);
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        server = new TestingSnowflakeServer();
+        if (!server.checkIfTableExists(TableType.ICEBERG, NATION)) {
+            executeOnSnowflake("""
+                    create or replace ICEBERG TABLE NATION (
+                    	NATIONKEY NUMBER(38,0),
+                    	NAME STRING,
+                    	REGIONKEY NUMBER(38,0),
+                    	COMMENT STRING
+                    )
+                     EXTERNAL_VOLUME = '%s'
+                     CATALOG = 'SNOWFLAKE'
+                     BASE_LOCATION = 'nation/'""".formatted(SNOWFLAKE_S3_EXTERNAL_VOLUME));
+
+            executeOnSnowflake("INSERT INTO NATION SELECT * FROM SNOWFLAKE_SAMPLE_DATA.TPCH_SF1.NATION");
```

Comment:
Is this schema supposed to pre-exist in Snowflake?

Reply:
As per https://docs.snowflake.com/en/user-guide/sample-data-using `SNOWFLAKE_SAMPLE_DATA` exists for newer accounts and there is a known way to create it if it doesn't.